cw_monero/lib/monero_wallet_addresses.dart

```diff
@@ -27,6 +29,40 @@ abstract class MoneroWalletAddressesBase extends WalletAddresses with Store {
   @observable
   String address;
 
+  @override
+  String get latestAddress {
+    final accountIndex = 0;
+    var addressIndex = subaddress_list.numSubaddresses(accountIndex) - 1;
+    var address = getAddress(accountIndex: accountIndex,addressIndex: addressIndex);
+    while (hiddenAddresses.contains(address)) {
+      addressIndex++;
+      address = getAddress(accountIndex: 0, addressIndex: addressIndex);
+    }
+    return address;
+  }
+
+  @override
+  String get addressForExchange {
+    final accountIndex = 0;
+    var addressIndex = subaddress_list.numSubaddresses(accountIndex) - 1;
+    var address = getAddress(accountIndex: accountIndex,addressIndex: addressIndex);
+    while (hiddenAddresses.contains(address) || manualAddresses.contains(address)) {
+      addressIndex++;
+      address = getAddress(accountIndex: 0, addressIndex: addressIndex);
+    }
+    return address;
+  }
+
+  String getLastAddress(int accountIndex) {
+    var addressIndex = subaddress_list.numSubaddresses(accountIndex) - 1;
+    var address = getAddress(accountIndex: accountIndex,addressIndex: addressIndex);
+    while (hiddenAddresses.contains(address)) {
```

Comment:
maybe I am misunderstanding what the lastAddress is all about, cuz you seem to be skipping the check for manualAddresses every time for the `lastAddress`

Reply:
Correct, when user creates a new address I assume that they want to use it - so I don't exclude them from the list